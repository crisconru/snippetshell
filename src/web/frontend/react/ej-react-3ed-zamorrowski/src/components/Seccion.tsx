import { useState } from 'react';

interface Props {
  texto?: string,
  fondo?: string,
  colorTexto?: string,
  children?: any
}

const CollapsableSeccion = ({texto, fondo, colorTexto, children}: Props) => {
  const [collapse, setCollapse] = useState(true)
  const toggleCollape = () => setCollapse(!collapse)
  return <>
    <h1 style={{backgroundColor: fondo, color: colorTexto}} onClick={toggleCollape}>{texto}</h1>
    <div style={{display: collapse ? 'none' : 'block'}}>
      {children}
    </div>
  </>
}

export const Clase = ({texto, children}: Props) => <CollapsableSeccion texto={texto} fondo='black' colorTexto='white' children={children}/>
export const Seccion = ({texto, children}: Props) => <CollapsableSeccion texto={texto} fondo='red' colorTexto='black' children={children}/>
export const Ejercicio = ({texto, children}: Props) => <CollapsableSeccion texto={texto} fondo='green' colorTexto='white' children={children}/>
