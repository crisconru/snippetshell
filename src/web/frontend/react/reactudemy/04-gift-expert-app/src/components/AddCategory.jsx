import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
    // console.log(`cambiada caja de texto con ${e.target.value}`)
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    if (inputValue.trim().length > 0) {
      setCategories((categories) => [inputValue, ...categories])
      setInputValue('')
    }
    // console.log('submit hecho')
  }

  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input type='text' value={inputValue} onChange={handleInputChange} />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
