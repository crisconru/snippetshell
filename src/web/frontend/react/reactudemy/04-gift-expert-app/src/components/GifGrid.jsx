import React from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {
  const { loading, data: images } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {loading && (
        <p className='animate__animated animate_flash'>Cargando...</p>
      )}
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
