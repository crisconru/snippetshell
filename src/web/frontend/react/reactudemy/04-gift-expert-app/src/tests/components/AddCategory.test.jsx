import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import { AddCategory } from '../../components/AddCategory'

describe('Pruebas para <AddCategory />', () => {
  const setCategories = jest.fn()
  let wrapper

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })
  /** /
  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input')
    input.simulate('change', { target: { value: 'Test AddCategory' } })
  })
  /**/
  test('No debe de postear la informacion con submit', () => {
    const e = { preventDefault() {} }
    wrapper.find('form').simulate('submit', e)
    expect(setCategories).not.toHaveBeenCalled()
  })

  test('Debe de llamar al setCategories y limpiar la caja de texto', () => {
    // Input - onChange
    const input = wrapper.find('input')
    const dataInput = { target: { value: 'hola mundo' } }
    input.simulate('change', dataInput)
    // Form - onSubmit
    const e = { preventDefault() {} }
    wrapper.find('form').simulate('submit', e)
    // Called
    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))
    // Result
    expect(input.prop('value')).toBe('')
  })
})
