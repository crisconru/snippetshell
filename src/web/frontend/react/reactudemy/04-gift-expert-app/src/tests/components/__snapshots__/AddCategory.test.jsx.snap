// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <AddCategory /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add Category
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
